src/components/ChooseOfferForm/ChooseOfferForm.tsx

```diff
@@ -24,107 +24,156 @@ type Props = {
   onBackButtonClickHandler?: () => void;
   monthlyOffer?: Offer;
   yearlyOffer?: Offer;
+  tvodOffers: Offer[];
   submitting: boolean;
 };
 
-const ChooseOfferForm: React.FC<Props> = ({ values, errors, onChange, onSubmit, submitting, yearlyOffer, monthlyOffer, onBackButtonClickHandler }: Props) => {
+type OfferBoxProps = {
+  offer: Offer;
+  radioKey: 'periodicity' | 'tvodOfferId';
+  radioValue: string;
+  title: string;
+  ariaLabel: string;
+  secondBenefit?: string;
+  periodString?: string;
+};
+type OfferBox = ({ offer, radioKey, radioValue, title, ariaLabel, secondBenefit, periodString }: OfferBoxProps) => JSX.Element;
+
+const ChooseOfferForm: React.FC<Props> = ({
+  values,
+  errors,
+  onChange,
+  onSubmit,
+  submitting,
+  yearlyOffer,
+  monthlyOffer,
+  tvodOffers,
+  onBackButtonClickHandler,
+}: Props) => {
   const { siteName } = useContext(ConfigContext);
   const { t } = useTranslation('account');
 
-  const getFreeTrialText = (offer: Offer) => {
-    if (offer.freeDays > 0) {
-      return t('choose_offer.benefits.first_days_free', { count: offer.freeDays });
-    } else if (offer.freePeriods) {
-      // t('periods.day')
-      // t('periods.week')
-      // t('periods.month')
-      // t('periods.year')
-      const period = t(`periods.${offer.period}`, { count: offer.freePeriods });
+  const getFreeTrialText = useCallback(
```

Comment:
`useCallback` is redundant here since it isn't passed as prop. When used in JSX directly, it is being called anyways.